listenbrainz/webserver/views/feedback_api.py

```diff
@@ -0,0 +1,74 @@
+import ujson
+import listenbrainz.db.user as db_user
+import listenbrainz.db.feedback as db_feedback
+
+from datetime import datetime
+from flask import Blueprint, current_app, jsonify, request
+from listenbrainz.webserver.decorators import crossdomain
+from listenbrainz.webserver.errors import (APIBadRequest,
+                                           APIInternalServerError, APINotFound,
+                                           APIServiceUnavailable,
+                                           APIUnauthorized)
+from listenbrainz.webserver.rate_limiter import ratelimit
+from listenbrainz.webserver.views.api import _validate_auth_header
+from listenbrainz.webserver.views.api_tools import log_raise_400, DEFAULT_ITEMS_PER_GET, MAX_ITEMS_PER_GET
+from listenbrainz.feedback import Feedback
+from pydantic import ValidationError
+
+#: Maximum overall feedback size in bytes, to prevent egregious spamming.
+MAX_FEEDBACK_SIZE = 10240
+
+feedback_api_bp = Blueprint('feedback_api_v1', __name__)
+
+@feedback_api_bp.route("submit-feedback", methods=["POST", "OPTIONS"])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def submit_feedback():
+    """
+    Submit recording feedback (love/hate) to the server. A user token (found on  https://listenbrainz.org/profile/ )
+    must be provided in the Authorization header! Each request should contain only one feedback in the payload.
+
+    Feedback should be submitted for tracks when the user has listened to them and they exist in their listen history.
+    If the user hasn't listened to the track or hasn't submitted it to ListenBrainz, it doesn't fully count as a feedback
+    and should not be submitted.
+
+    For complete details on the format of the JSON to be POSTed to this endpoint, see :ref:`feedback-json-doc`.
+
+    :reqheader Authorization: Token <user token>
+    :statuscode 200: feedback accepted.
+    :statuscode 400: invalid JSON sent, see error message for details.
+    :statuscode 401: invalid authorization. See error message for details.
+    :resheader Content-Type: *application/json*
+    """
+    user = _validate_auth_header()
+
+    raw_data = request.get_data()
```

Comment:
The JSON being posted to the endpoint is simple enough that I think we should use request.json and leave the parsing of json to flask. I think you can drop all the code from here until line 55 and use the built in json support.

Reply:
Removed